### ISSUE TITLE
[switch]: Should author defined content be allowed on tracks or thumb?

### ISSUE THREAD

**Issue by gfellerph** [openui/open-ui#959]
I'm working on the [switch proposal pull request](https://github.com/openui/open-ui/pull/785). @scottaohara raised multiple concerns with author defined content inside the different parts of the proposed `<switch>` element.

### Question
Should authors be able to add text content to either the `track` or the `thumb` elements?

### Research
About a third of analyzed Design Systems use this feature ([Design System Analysis section in the pull request](https://github.com/openui/open-ui/pull/785/files#diff-2588d6f04d5c0446fa428f7c315468c4680338ffbefffcfb61b659f2a3367d51R445)).
- 30% display content on the track
- 18% display content on the thumb

<img width="69" alt="image" src="https://github.com/openui/open-ui/assets/1659006/966f8f99-aed5-4070-96cc-db3081bd31e2">

Example: https://ant.design/components/switch

### Accessibility concerns
- Content mapping will be difficult since it's a new concept
- The switch is already described by its label and content should be ignored

_Relevant comment: https://github.com/openui/open-ui/pull/785#discussion_r1396420527_

### Layout issues
Custom content can have various layout issues.

- The content on both sides cannot be wider than the thumb. Otherwise, the thumb is no longer capable of hiding the content of the inactive track content. ![image](https://github.com/openui/open-ui/assets/1659006/7ef9c396-81d6-4b7b-8c19-8a937d2476f0)
- It’s unclear how to style the track and slider if the track content can be of arbitrary length ![image](https://github.com/openui/open-ui/assets/1659006/962599db-284d-461a-bcb7-f6c1304aed92)

### Possible solutions
1. Allow for author defined text/content and find solutions to the above issues
2. Don't allow for author defined text and risk not adressing all existing use cases
3. Only allow decorative images or svg content, which covers most use cases and avoids the above issues mostly
4. ...?

**Comment by lukewarlow** [comment]
My initial reaction is I think we should allow content(image/SVG is probably enough but enforcing this seems difficult?) in the thumbs (but this should be hidden from screen readers) I'm less sure on use cases for tracks.

**Comment by brechtDR** [comment]
After the Telecon, i did found another example where a switch could mean something else than on/off in a codepen.
how about am/pm? (Yes, i was fishing for cookies)


https://codepen.io/bnthor/pen/WQBNxO

**Comment by scottaohara** [comment]
@brechtDR that's really more of a radio button then that has been restyled to look like a switch.

i see stuff like this a lot as well, where people "want" the visual design of one type of control, but they style it to resemble a different control type - or they use the control they want in a way that distorts its intended use.

**Comment by brechtDR** [comment]
@scottaohara i agree with this. My question is rather if people won't be looking to use it for those purposes. It's a hard path in this case: design vs function. But maybe I don't really understand the big need for this and am just biased as I've been styling these things with checkboxes too much. Maybe I'm a bit "scared" that the use case and need is too specific or limited. But that is subjective and thus a personal opinion of course. I can understand other people want this in its purest on/off-but not a checkbox way.

**Comment by lukewarlow** [comment]
https://twitter.com/jh3yy/status/1733296190244979015?t=f_lc95VlAVjp20jZBTXpNg&s=19

Maybe a benchmark of crazy design to see if any proposal can manage 😅

**Comment by gfellerph** [comment]
@brechtDR another example like this, from the Swiss railways (https://www.sbb.ch/) for the departure/arrival control (at least this one is a radiogroup, but looks like a switch).

@lukewarlow awesome example, I was looking for this to reference in the proposal but could not find it anymore.

I'm wondering if this would be something that could be made easier for devs by adding accessibility bindings to a switch which allows content on the track.

```html
<label for="timetable">Show timetable for</label>
<switch id="timetable">
  <track>
    <trackstart><abbr title="Departure">Dep</abbr></trackstart>
    <trackend><abbr title="Arrival">Arr</abbr></trackend>
  </track>
</switch>
```

Which could read something like "Departure, selected, switch, 1 of 2". @scottaohara, would this even make sense?

This, however, would strongly contradict the idea that the switch is a binary state of a single value.

**Comment by lukewarlow** [comment]
https://github.com/whatwg/html/pull/9546#issuecomment-1865357407 - The HTML proposal from Apple has an implementation to try which might be useful. Currently it supports content attribute on tracks and thumbs
